app/src/main/java/org/oppia/android/app/application/OppiaApplication.kt

```diff
@@ -15,27 +15,50 @@ class OppiaApplication :
   ActivityComponentFactory,
   ApplicationInjectorProvider,
   Configuration.Provider {
+
+  /** Boolean variable to enable dev mode. It will decide which application component to use */
+  private val enableDevMode = true
+
   /** The root [ApplicationComponent]. */
   private val component: ApplicationComponent by lazy {
     DaggerApplicationComponent.builder()
       .setApplication(this)
       .build()
   }
 
+  /** The root [ApplicationComponentForDevMode]. */
+  private val componentForDevMode: ApplicationComponentForDevMode by lazy {
+    DaggerApplicationComponentForDevMode.builder()
+      .setApplication(this)
+      .build()
+  }
+
   override fun createActivityComponent(activity: AppCompatActivity): ActivityComponent {
-    return component.getActivityComponentBuilderProvider().get().setActivity(activity).build()
+    return if (enableDevMode)
```

Comment:
It might be worth mentioning what we're doing here in a comment.

Reply:
No longer used.